plugins/modules/files/sapcar_extract.py

```diff
@@ -0,0 +1,220 @@
+        if os.path.isdir(fullPath):
+            allFiles = allFiles + [fullPath]
+            allFiles = allFiles + getListOfFiles(fullPath)
+        else:
+            allFiles.append(fullPath)
+    return allFiles
+
+
+def downloadSAPCAR(binary_path):
+    bin_path = None
+    # download sapcar binary if url is provided otherwise path is returned
+    if binary_path is not None:
+        try:
+            with open_url(binary_path) as response:
+                with open("/tmp/sapcar", 'wb') as out_file:
+                    data = response.read()
+                    out_file.write(data)
+            os.chmod("/tmp/sapcar", 0o755)
+            bin_path = "/tmp/sapcar"
+        except Exception:
+            bin_path = binary_path
+    return bin_path
+
+
+def checkifPresent(command, path, dest, signature, manifest):
+    # manipuliating output from SAR file for compare with already extracted files
+    iter_command = [command, '-tvf', path]
+    sar_out = module.run_command(iter_command)[1]
+    sar_raw = sar_out.split("\n")[1:]
+    if dest[-1] != "/":
+        dest = dest + "/"
+    sar_files = [dest + x.split(" ")[-1] for x in sar_raw if x]
+    # remove any SIGNATURE.SMF from list because it will not unpacked if signature is false
+    if not signature:
+        sar_files = [item for item in sar_files if '.SMF' not in item]
+    # if signature is renamed manipulate files in list of sar file for compare.
+    if manifest != "SIGNATURE.SMF":
+        sar_files = [item for item in sar_files if '.SMF' not in item]
+        sar_files = sar_files + [manifest]
+    # get extracted files if present
+    files_extracted = getListOfFiles(dest)
+    # compare extracted files with files in sar file
+    present = all(elem in files_extracted for elem in sar_files)
+    return present
+
+
+def main():
+    global module
+
+    module = AnsibleModule(
+        argument_spec=dict(
+            path=dict(type='path', required=True),
+            dest=dict(type='path'),
+            binary_path=dict(type='path'),
+            signature=dict(type='bool', default=False),
+            manifest=dict(type='str', default="SIGNATURE.SMF"),
+            remove=dict(type='bool', default=False),
+        ),
+        supports_check_mode=True,
+    )
+    rc, out, err = [0, "", ""]
+    params = module.params
+    check_mode = module.check_mode
+
+    path = params['path']
+    dest = params['dest']
+    signature = params['signature']
+    manifest = params['manifest']
+    remove = params['remove']
+
+    bin_path = downloadSAPCAR(params['binary_path'])
+
+    if dest is None:
+        dest_head_tail = os.path.split(path)
+        dest = dest_head_tail[0] + '/'
+    else:
+        if not os.path.exists(dest):
+            os.makedirs(dest, 0o755)
+
+    if bin_path is not None:
+        command = [module.get_bin_path(bin_path, required=True)]
+    else:
+        try:
+            command = [module.get_bin_path('/tmp/sapcar' or '/tmp/SAPCAR', required=True)]
```

Comment:
Looking in `/tmp` is definitely **not** a good idea. This opens a can of security issues.

Reply:
that is true. I removed it and raise an error if it is not available